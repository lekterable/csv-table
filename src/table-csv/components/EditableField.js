import React from 'react'

export default ({
  value = '',
  submitContent,
  onChange,
  onKeyDown,
  toggleMode,
  editMode
}) => {
  return editMode ? (
    <input
      data-testid="editable-field-input"
      type="text"
      className="editable-field editable-field--input"
      readOnly={!onChange}
      value={value}
      onBlur={submitContent}
      onChange={onChange}
      onKeyDown={onKeyDown}
      autoFocus
      size={value.length + 1}
    />
  ) : (
    <div
      data-testid="editable-field-display"
      className="editable-field"
      onClick={toggleMode}
    >
      {value}
    </div>
  )
}
