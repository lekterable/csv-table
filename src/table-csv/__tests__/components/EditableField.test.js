import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import EditableField from '../../components/EditableField'

describe('<EditableField />', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const component = renderer.create(<EditableField />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should render display', () => {
    const { queryByTestId } = render(<EditableField />)
    const field = queryByTestId('editable-field-display')
    const input = queryByTestId('editable-field-input')

    expect(input).toBe(null)
    expect(field).not.toBe(null)
  })

  it('should render input if `editMode` prop', () => {
    const { queryByTestId } = render(<EditableField editMode />)
    const field = queryByTestId('editable-field-display')
    const input = queryByTestId('editable-field-input')

    expect(field).toBe(null)
    expect(input).not.toBe(null)
  })

  it('should render correct value', () => {
    const { queryByTestId } = render(<EditableField value="Just a field" />)
    const field = queryByTestId('editable-field-display')

    expect(field.textContent).toBe('Just a field')
  })
})
