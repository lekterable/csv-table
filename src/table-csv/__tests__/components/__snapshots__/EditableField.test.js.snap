// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableField /> should match snapshot 1`] = `
<div
  className="editable-field"
  data-testid="editable-field-display"
>
  
</div>
`;
